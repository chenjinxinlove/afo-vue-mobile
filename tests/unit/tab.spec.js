import {mount} from '@vue/test-utils'
import Demo from '../../src/tab/demo'
import Vue from 'vue'
import Tab from '../../src/tab'
import Tabs from '../../src/tabs/index'

describe('Tag.vue', () => {
  // 测试是否可以被使用
  it('use', () => {
    Vue.use(Tab)
    expect(Vue.component(Tab.name))
      .toHaveProperty('name', 'VueComponent')
  })
  it('use', () => {
    Vue.use(Tabs)
    expect(Vue.component(Tabs.name))
      .toHaveProperty('name', 'VueComponent')
  })
  // demo测试快照
  it('demo snapshot', () => {
    expect(mount(Demo)).toMatchSnapshot()
  })
})
