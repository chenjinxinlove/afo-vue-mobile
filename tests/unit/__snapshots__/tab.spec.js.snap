// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag.vue demo snapshot 1`] = `
<div>
  <p>默认显示</p>
  <div class="afo-tabs--line afo-tabs" sticky="">
    <div class="afo-hairline--top-bottom afo-tabs__wrap afo-tabs__wrap--position">
      <div class="afo-tabs__nav afo-tabs__nav--line">
        <div class="afo-tabs__line"></div>
      </div>
    </div>
    <div class="afo-tabs__content">
      <div class="afo-tab__pane" style="display: none;">
        <!---->
        <!---->
      </div>
      <div class="afo-tab__pane" style="display: none;">
        <!---->
        <!---->
      </div>
      <div class="afo-tab__pane" style="display: none;">
        <!---->
        <!---->
      </div>
      <div class="afo-tab__pane" style="display: none;">
        <!---->
        <!---->
      </div>
    </div>
  </div>
  <p>显示card类型</p>
  <div class="afo-tabs--card afo-tabs">
    <div class="afo-tabs__wrap afo-tabs__wrap--position">
      <div class="afo-tabs__nav afo-tabs__nav--card">
        <!---->
      </div>
    </div>
    <div class="afo-tabs__content">
      <div class="afo-tab__pane" style="display: none;">
        <!---->
        <!---->
      </div>
      <div class="afo-tab__pane" style="display: none;">
        <!---->
        <!---->
      </div>
      <div class="afo-tab__pane" style="display: none;">
        <!---->
        <!---->
      </div>
      <div class="afo-tab__pane" style="display: none;">
        <!---->
        <!---->
      </div>
    </div>
  </div>
  <p>禁用类型</p>
  <div class="afo-tabs--line afo-tabs">
    <div class="afo-hairline--top-bottom afo-tabs__wrap afo-tabs__wrap--position">
      <div class="afo-tabs__nav afo-tabs__nav--line">
        <div class="afo-tabs__line"></div>
      </div>
    </div>
    <div class="afo-tabs__content">
      <div class="afo-tab__pane" style="display: none;">
        <!---->
        <!---->
      </div>
      <div class="afo-tab__pane" style="display: none;">
        <!---->
        <!---->
      </div>
      <div class="afo-tab__pane" style="display: none;">
        <!---->
        <!---->
      </div>
      <div class="afo-tab__pane" style="display: none;">
        <!---->
        <!---->
      </div>
    </div>
  </div>
  <p>点击事件</p>
  <div class="afo-tabs--line afo-tabs">
    <div class="afo-hairline--top-bottom afo-tabs__wrap afo-tabs__wrap--position">
      <div class="afo-tabs__nav afo-tabs__nav--line">
        <div class="afo-tabs__line"></div>
      </div>
    </div>
    <div class="afo-tabs__content">
      <div class="afo-tab__pane" style="display: none;">
        <!---->
        <!---->
      </div>
      <div class="afo-tab__pane" style="display: none;">
        <!---->
        <!---->
      </div>
      <div class="afo-tab__pane" style="display: none;">
        <!---->
        <!---->
      </div>
      <div class="afo-tab__pane" style="display: none;">
        <!---->
        <!---->
      </div>
    </div>
  </div>
  <p>横向滑动</p>
  <div class="afo-tabs--line afo-tabs">
    <div class="afo-hairline--top-bottom afo-tabs__wrap afo-tabs__wrap--position">
      <div class="afo-tabs__nav afo-tabs__nav--line">
        <div class="afo-tabs__line"></div>
      </div>
    </div>
    <div class="afo-tabs__content">
      <div class="afo-tab__pane" style="display: none;">
        <!---->
        <!---->
      </div>
      <div class="afo-tab__pane" style="display: none;">
        <!---->
        <!---->
      </div>
      <div class="afo-tab__pane" style="display: none;">
        <!---->
        <!---->
      </div>
      <div class="afo-tab__pane" style="display: none;">
        <!---->
        <!---->
      </div>
      <div class="afo-tab__pane" style="display: none;">
        <!---->
        <!---->
      </div>
      <div class="afo-tab__pane" style="display: none;">
        <!---->
        <!---->
      </div>
      <div class="afo-tab__pane" style="display: none;">
        <!---->
        <!---->
      </div>
      <div class="afo-tab__pane" style="display: none;">
        <!---->
        <!---->
      </div>
    </div>
  </div>
  <p>滑动切换</p>
  <div class="afo-tabs--line afo-tabs">
    <div class="afo-hairline--top-bottom afo-tabs__wrap afo-tabs__wrap--position">
      <div class="afo-tabs__nav afo-tabs__nav--line">
        <div class="afo-tabs__line"></div>
      </div>
    </div>
    <div class="afo-tabs__content">
      <div class="afo-tab__pane" style="display: none;">
        <!---->
        <!---->
      </div>
      <div class="afo-tab__pane" style="display: none;">
        <!---->
        <!---->
      </div>
      <div class="afo-tab__pane" style="display: none;">
        <!---->
        <!---->
      </div>
      <div class="afo-tab__pane" style="display: none;">
        <!---->
        <!---->
      </div>
    </div>
  </div>
</div>
`;
